src/core/anchor-expander.js

```diff
@@ -0,0 +1,141 @@
+        break;
+      }
+      case "figure": {
+        processFigure(matchingElement, id, a);
+        break;
+      }
+      case "aside":
+      case "div": {
+        processBox(matchingElement, id, a);
+        break;
+      }
+      default: {
+        a.textContent = a.getAttribute("href");
+        const msg = "ReSpec doesn't support expanding this kind of reference.";
+        showInlineError(a, msg, `Can't expand "#${id}".`);
+      }
+    }
+    localize(matchingElement, a);
+    a.normalize();
+  }
+}
+
+function processBox(matchingElement, id, a) {
+  const selfLink = matchingElement.querySelector(".marker .self-link");
+  if (!selfLink) {
+    a.textContent = a.getAttribute("href");
+    const msg = `Found matching element "${id}", but it has no title or marker.`;
+    showInlineError(a, msg, "Missing title.");
+    return;
+  }
+  const children = makeSafeCopy(selfLink);
+  a.append(...children);
+  a.classList.add("box-ref");
+}
+
+function processFigure(matchingElement, id, a) {
+  const figcaption = matchingElement.querySelector("figcaption");
+  if (!figcaption) {
+    a.textContent = a.getAttribute("href");
+    const msg = `Found matching figure "${id}", but figure is lacking a \`<figcaption>\`.`;
+    showInlineError(a, msg, "Missing figcaption in referenced figure.");
+    return;
+  }
+  // remove the figure's title
+  const children = [...makeSafeCopy(figcaption)].filter(
+    node => !node.classList || !node.classList.contains("fig-title")
+  );
+  // drop an empty space at the end.
+  children.pop();
+  a.append(...children);
+  a.classList.add("fig-ref");
+  const figTitle = figcaption.querySelector(".fig-title");
+  if (!a.hasAttribute("title") && figTitle) {
+    a.title = norm(figTitle.textContent);
+  }
+}
+
+function processSection(matchingElement, id, a) {
+  const heading = matchingElement.querySelector("h6, h5, h4, h3, h2");
+  if (!heading) {
+    a.textContent = a.getAttribute("href");
+    const msg =
+      "Found matching section, but the section was lacking a heading element.";
+    showInlineError(a, msg, `No matching id in document: "${id}".`);
+    return;
+  }
+  processHeading(heading, a);
+  localize(heading, a);
+}
+
+function processHeading(heading, a) {
+  const hadSelfLink = heading.querySelector(".self-link");
+  const children = [...makeSafeCopy(heading)].filter(
+    node => !node.classList || !node.classList.contains("self-link")
+  );
+  a.append(...children);
+  if (hadSelfLink) a.prepend("§\u00A0");
+  a.classList.add("sec-ref");
+}
+
+function makeSafeCopy(node) {
+  const clone = node.cloneNode(true);
+  clone.querySelectorAll("[id]").forEach(elem => elem.removeAttribute("id"));
+  clone.querySelectorAll("dfn").forEach(dfn => {
+    const span = document.createElement("span");
+    span.append(...dfn.childNodes);
+    [...dfn.attributes].forEach(({ name, value }) => {
+      span.setAttribute(name, value);
+    });
+    dfn.parentNode.replaceChild(span, dfn);
```

Comment:
The whole thing in this loop could be replaced with `renameElement()` from `core/utils.js`.

Reply:
oh! Forgot about that! thanks!!!